test_common/harness/imageHelpers.cpp

```diff
@@ -912,8 +912,6 @@ void get_max_sizes(
 float get_max_absolute_error(const cl_image_format *format,
                              image_sampler_data *sampler)
 {
-    if (sampler->filter_mode == CL_FILTER_NEAREST) return 0.0f;
```

Comment:
I tried to follow what this test is doing but it wasn't immediately apparent to me.

Do we know why the absolute error was forced to zero previously for NEAREST filtering?  Why is it correct to remove this check?  Removing the check means we'll have looser error bounds for formats beyond just CL_UNORM_SHORT_555 and CL_UNORM_SHORT_565.

Reply:
I have not used cl images that much, so I might be wrong, but here is what I think:

The `filter_mode` allows you to have different strategies on which elements of the image to consider when the coordinates used to access are floating point coordinates.

Independently from the final coordinates chosen, an absolute error is created because of normalization:
Let's take `CL_UNORM_SHORT_555` for example
You start with a floating point value in the range `[0.0, 1.0]` , this value will be stored using 5bit (so in the integer range `[0, 31]`). During this first transformation, some data will be lost.

This is the same with all normalized formats.
Even when the `filter_mode` is `CL_FILTER_NEAREST` this error exists.

Reply:
I have the same concern as @bashbaug. The motivation for this whole function is unclear, but my best guess is that it's intended to allow an arbitrary amount of extra error for normalized types when in CL_FILTER_LINEAR mode. Removing the LINEAR conditional drastically loosens the requirements for all normalized types when in NEAREST mode so I don't think we should make that change.

As Ben mentioned, it would be helpful to get background on which test cases are fixed by this change.